src/main/groovy/grails/plugin/sentry/SpringSecurityUserEventBuilderHelper.groovy

```diff
@@ -0,0 +1,62 @@
+package grails.plugin.sentry
+
+import com.getsentry.raven.event.EventBuilder
+import com.getsentry.raven.event.helper.EventBuilderHelper
+import com.getsentry.raven.event.interfaces.UserInterface
+
+import javax.servlet.http.HttpServletRequest
+
+/**
+ * @author <a href='mailto:alexey@zhokhov.com'>Alexey Zhokhov</a>
+ */
+class SpringSecurityUserEventBuilderHelper implements EventBuilderHelper {
+
+    static List<String> ipHeaders = ['X-Real-IP',
+                                     'Client-IP',
+                                     'X-Forwarded-For',
+                                     'Proxy-Client-IP',
+                                     'WL-Proxy-Client-IP',
+                                     'rlnclientipaddr']
+
+    def springSecurityService
+    def ravenServletRequestListener
+
+    @Override
+    void helpBuildingEvent(EventBuilder eventBuilder) {
+        def currentUser = springSecurityService?.getCurrentUser()
```

Comment:
That's a good  point - it would be better if possible to retrieve data from the `principal`, which would be a `GrailsUser` if authenticated, so the username, user domain class id, authority names, and the various locked/enabled booleans would be available without making an unnecessary database call. Call `springSecurityService.isLoggedIn()` first (to verify there's an `Authentication` and check that it's not anonymous), and then `springSecurityService.getPrincipal()`

The only property that's not already available in GrailsUser is the email property, but as Eric pointed out that's nonstandard and would need to be added the domaain class in many apps, and while doing that they could also configure their custom `UserDetailsService` and `GrailsUser` subclass.